src/components/common/validation/time-format/ValidatedTimeEntry.jsx

```diff
@@ -9,32 +11,26 @@ const ValidatedTimeEntry = ({
   defaultValue,
   register,
   isRequired,
-  formState,
+  getValues,
+  timeEntry,
+  timeEntriesIndex,
 }) => {
   const errorMessage = `You must enter a ${timeType} in the HH:MM 24 hour clock format`;
 
-  useEffect(() => {
-    isFinishTimeNextDay();
-  }, [formState]);
+  const { timeEntries, setTimeEntries } = useTimecardContext();
 
-  const isFinishTimeNextDay = () => {
-    if (document.getElementById(`shift-finish-time`)) {
-      var finishTimeValue = document.getElementById(`shift-finish-time`).value;
-    }
-    if (document.getElementById(`shift-start-time`)) {
-      var startTimeValue = document.getElementById(`shift-start-time`).value;
-    }
-    var answer = calculateFinishTimeOnNextDay(startTimeValue, finishTimeValue);
+  const setFinishTimeText = () => {
+    const startTimeValue = getValues('shift-start-time');
+    const finishTimeValue = getValues('shift-finish-time');
+    const newTimeEntries = deepCloneJson(timeEntries);
 
-    if (document.getElementById('end-next-day')) {
-      if (answer >= 1) {
-        document.getElementById('end-next-day').innerHTML =
-          'Finishes on next day';
-      } else {
-        document.getElementById('end-next-day').innerHTML = '';
-      }
-    }
+    let answer = isFinishTimeOnNextDay(startTimeValue, finishTimeValue);
```

Comment:
This should be declared as const as we won't be changing it value, and also needs a meaningful name.
Or we could just pass this function call directly into line 30.

Reply:
updated